Client/Frontend/Settings/HomepageSettings/WallpaperSettings/v2/WallpaperSettingsViewModel.swift

```diff
@@ -0,0 +1,210 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+public enum WallpaperSettingsError: Error {
+    case itemNotFound
+}
+
+class WallpaperSettingsViewModel {
+
+    enum WallpaperSettingsLayout: Equatable {
+        case compact
+        case regular
+
+        // The maximum number of items to display per row
+        var itemsPerRow: Int {
+            switch self {
+            case .compact: return 3
+            case .regular: return 4
+            }
+        }
+    }
+
+    struct Constants {
+        struct Strings {
+            struct Toast {
+                static let label: String = .Settings.Homepage.Wallpaper.WallpaperUpdatedToastLabel
+                static let button: String = .Settings.Homepage.Wallpaper.WallpaperUpdatedToastButton
+            }
+        }
+    }
+
+    private var wallpaperManager: WallpaperManagerInterface
+    private var wallpaperCollections = [WallpaperCollection]()
+    var tabManager: TabManager
+    var sectionLayout: WallpaperSettingsLayout = .compact // We use the compact layout as default
+    var selectedIndexPath: IndexPath = IndexPath(row: 0, section: 0)
+
+    var numberOfSections: Int {
+        return wallpaperCollections.count
+    }
+
+    init(wallpaperManager: WallpaperManagerInterface = WallpaperManager(), tabManager: TabManager) {
+        self.wallpaperManager = wallpaperManager
+        self.tabManager = tabManager
+        setupWallpapers()
+    }
+
+    func numberOfWallpapers(in section: Int) -> Int {
+        return wallpaperCollections[safe: section]?.wallpapers.count ?? 0
+    }
+
+    func sectionHeaderViewModel(for sectionIndex: Int, dismissView: @escaping (() -> Void)) -> WallpaperSettingsHeaderViewModel? {
+        guard let collection = wallpaperCollections[safe: sectionIndex] else { return nil }
+
+        let isClassic = collection.type == .classic
+        let title: String = isClassic ?
+            .Settings.Homepage.Wallpaper.ClassicWallpaper : .Settings.Homepage.Wallpaper.LimitedEditionWallpaper
```

Comment:
For readability & brevity, you could always typealias `String.Settings.Homepage.Wallpaper`.